Graphics/GraphicsEngineOpenGL/CMakeLists.txt

```diff
@@ -213,10 +213,20 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         )
     endif()
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    set_target_properties(Diligent-GraphicsEngineOpenGL-shared PROPERTIES
-        # Disallow missing direct and indirect dependencies to enssure that .so is self-contained
-        LINK_FLAGS "-Wl,--no-undefined -Wl,--no-allow-shlib-undefined"
-    )
+    if(PLATFORM_WIN32)
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/export.map
+            "{ global: *GetEngineFactoryOpenGL*; local: *; };"
+        )
+        set_target_properties(Diligent-GraphicsEngineOpenGL-shared PROPERTIES
+            # Disallow missing direct and indirect dependencies to enssure that .so is self-contained
+            LINK_FLAGS "-Wl,--no-undefined -Wl,--no-allow-shlib-undefined -Wl,--version-script=export.map"
+        )
```

Comment:
Will this work if you just have this win32-specific branch to eliminate duplication:
```
if(PLATFORM_WIN32)
    # Please write a comment about what this does
    file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/export.map
          "{ global: *GetEngineFactoryOpenGL*; local: *; };"
    )
    set_target_properties(Diligent-GraphicsEngineOpenGL-shared PROPERTIES
           LINK_FLAGS "-Wl,--version-script=export.map"
    )
endif()
```